metakb/harvesters/pmkb.py

```diff
@@ -0,0 +1,200 @@
+        """
+        try:
+            self.pmkb_dir = PROJECT_ROOT / 'data' / 'pmkb'
+            self._check_files()
+            variants = self._get_all_variants()
+            statements = self._get_all_interpretations(variants)
+            self._create_json(statements, variants, fn)
+        except NotImplementedError:  # noqa: E722
+            logger.error("PMKB Harvester was not successful.")
+            return False
+
+    def _check_files(self):
+        """Check PMKB data directory for requisite files and call download
+        methods if needed.
+        """
+        self.pmkb_dir.mkdir(exist_ok=True, parents=True)
+
+        if not list(self.pmkb_dir.glob('pmkb_variants_*.csv')) and \
+                not list(self.pmkb_dir.glob('pmkb_interps_*.csv')):
+            self._download_data()
+
+    def _download_data(self):
+        """Retrieve PMKB data from remote host."""
+        logging.info("Downloading PMKB source data...")
+        self.pmkb_dir.mkdir(exist_ok=True, parents=True)
+        version = datetime.now().strftime('%Y%m%d')
+
+        response_interps = requests.get('http://pmkb.org/therapies/download.csv')  # noqa: E501
+        if response_interps.status_code == 200:
+            fname = self.pmkb_dir / f'pmkb_interps_{version}.csv'
+            with open(fname, 'wb') as f:
+                f.write(response_interps.content)
+        else:
+            msg = 'PMKB interpretations CSV download failed.'
+            logger.error(msg)
+            raise requests.exceptions.RequestException(msg)
+
+        response_variants = requests.get('http://pmkb.org/variants/download.csv')  # noqa: E501
+        if response_variants.status_code == 200:
+            fname = self.pmkb_dir / f'pmkb_variants_{version}.csv'
+            with open(fname, 'wb') as f:
+                f.write(response_variants.content)
+        else:
+            msg = 'PMKB interpretations CSV download failed.'
+            logger.error(msg)
+            raise requests.exceptions.RequestException(msg)
+        logging.info("PMKB source data downloads complete.")
+
+    def _get_all_variants(self):
+        """Process PMKB variants.
+        :return: Dict keying variant names (string) to data objects
+        """
+        pattern = 'pmkb_variants_*.csv'
+        variants_path = sorted(list(self.pmkb_dir.glob(pattern)))[-1]
+        variants_file = open(variants_path, 'r')
+        reader = csv.DictReader(variants_file)
+        variants = {}
+
+        for variant in reader:
+            name = variant['Description']
+            if name in variants:
+                logger.error(f"Multiple records for variant: {name}")
+                continue
+            variant_id = variant['PMKB URL'].split('/')[-1]
+
+            variant_object = {
+                "name": name,
+                "gene": {
+                    "name": variant['Gene'],
+                },
+                "id": variant_id,
+                "origin": variant['Germline/Somatic'].lower(),
```

Comment:
Not a huge deal, but since we're just harvesting, should we be altering the data?